Ryujinx.Graphics.Gpu/Engine/GPFifo/GPFifoDevice.cs

```diff
@@ -163,7 +163,7 @@ public void PushEntries(ReadOnlySpan<ulong> entries)
         /// <returns>True if commands were received, false if wait timed out</returns>
         public bool WaitForCommands()
         {
-            return _event.WaitOne(8);
+            return _event.WaitOne(8) && _commandBufferQueue.Count > 0;
```

Comment:
Perhaps?
```suggestion
            return _event.WaitOne(8) && !_commandBufferQueue.IsEmpty;
```
AFAIK `ConcurrentQueue<T>.IsEmpty` is cheaper than `ConcurrentQueue<T>.Count > 0`.